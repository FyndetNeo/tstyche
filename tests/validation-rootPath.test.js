import { afterEach, describe, expect, test } from "@jest/globals";
import { clearFixture, getFixtureUrl, writeFixture } from "./__utils__/fixtureFactory.js";
import { normalizeOutput } from "./__utils__/normalizeOutput.js";
import { spawnTyche } from "./__utils__/spawnTyche.js";

const fixtureUrl = getFixtureUrl("validation-rootPath", { generated: true });

afterEach(async () => {
  await clearFixture(fixtureUrl);
});

describe("'rootPath' configuration file option", () => {
  test("when specified path does not exist", async () => {
    const config = {
      rootPath: "../nope",
    };

    await writeFixture(fixtureUrl, {
      ["config/tstyche.json"]: JSON.stringify(config, null, 2),
    });

    const { exitCode, stderr, stdout } = await spawnTyche(fixtureUrl, ["--config ./config/tstyche.json"]);

    expect(stdout).toBe("");
    expect(normalizeOutput(stderr)).toMatchSnapshot();

    expect(exitCode).toBe(1);
  });
});
