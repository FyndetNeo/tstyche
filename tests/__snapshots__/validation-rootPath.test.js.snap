// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'rootPath' configuration file option when specified path does not exist: stderr 1`] = `
"Error: The specified path '<<cwd>>/tests/__fixtures__/.generated/validation-rootPath/nope' does not exist.

  1 | {
> 2 |   "rootPath": "../nope"
    |               ^
  3 | }

      at ./config/tstyche.json:2:15

"
`;
